### PR TITLE
Updated Cell Range to Accept Numerical Only and Letter Only Inputs

### LINKED ISSUE #25: Cell Address does not accept numerical only cell addresses
https://github.com/BHoM/Excel_Toolkit/blob/develop/Excel_Engine/Query/IsValid.cs#L85

This method is failing to allow a Cell Address that looks like `2:2` - without any letters - despite this being a valid cell range in Excel.

It is also failing to allow a Cell Address that looks like `A:A` - without any numbers - despite this being a valid cell range in Excel.

### PR DESCRIPTION
**Pull request by desaiwangBH**
### Issues addressed by this PR

Closes #25 

-Updated IsValidRange to accept letters-only and numbers-only inputs in addition to standard cell addresses, for example A:A, 2:3.
-Also updated regular expressions for cell addresses and column addresses to be case insensitive.

### CODE REVIEW

**Comment by travispotterBH** [comment]
Let's provide a test file. Can you build something up in Grasshopper? @desaiwangBH

File: Excel_Engine/Query/IsValid.cs [hunk 1]
```diff
@@ -119,27 +119,42 @@ public static bool IsValidRange(this string range)
                 return false;
             }
 
-            if (!m_RangeFormat.IsMatch(range))
+            string[] split = range.Split(new char[] { ':' });
+
+            if (split.Length != 2)
             {
-                BH.Engine.Base.Compute.RecordError($"Range equal to {range} is not valid: it needs to consist of two sets of capital letters followed by digits, divided with a colon, for example 'A1:Z99'.");
+                BH.Engine.Base.Compute.RecordError("Must provided values in the format of 'expression1:expression2'");
```
Comment (FraserGreenroyd) [inline Excel_Engine/Query/IsValid.cs:126]:
```suggestion
                BH.Engine.Base.Compute.RecordError("Range values must be in the format of 'expression1:expression2'");
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: Excel_Engine/Query/IsValid.cs [hunk 2]
```diff
@@ -119,27 +119,42 @@ public static bool IsValidRange(this string range)
...
-            string[] split = range.Split(new char[] { ':' });
             string from = split[0];
             string to = split[1];
 
-            return from.IsValidAddress() && to.IsValidAddress();
+            Regex matchBegins0 = new Regex("^0$");
+
+            if (matchBegins0.IsMatch(from) || matchBegins0.IsMatch(to))
+            {
+                BH.Engine.Base.Compute.RecordError("Cell value cannot be or begin with 0.")
```
Comment (FraserGreenroyd) [inline Excel_Engine/Query/IsValid.cs:137]:
```suggestion
                BH.Engine.Base.Compute.RecordError("Cell value cannot be or begin with 0.");
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Review by FraserGreenroyd — APPROVED** [review]
Tested using SAP workflow script from @EKAdebo and is working nicely. Great work team!

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**